lib/dpdk/iceBoardVDIF.hpp

```diff
@@ -291,7 +291,13 @@ void iceBoardVDIF::copy_packet_vdif(struct rte_mbuf* mbuf) {
     int from_idx = header_offset + offset;
     int mbuf_len = mbuf->data_len;
     for (uint32_t time_step = 0; time_step < samples_per_packet; ++time_step) {
+        int output_idx_base = (vdif_frame_location * frame_size +           // Frame location in output buffer.
+			       vdif_header_len +        // Offset for the vdif header.
+			       tel.to_freq_id(encoded_id, 0) + // Offset for ARO as in tel.to_freq_id for _num_freq_per_stream = 128
+			       vdif_packet_len * num_elements * time_step); // Time step in output frame.
         for (int freq = 0; freq < 128; ++freq) {
+	    int output_idx = (output_idx_base +
+			      freq * 8); // Location in the VDIF packet is just frequency.
             for (int elem = 0; elem < num_elements; ++elem) {
 
                 // Advance to the next mbuf in the chain.
```

Comment:
@andrerenard - I don't quite know the structure of these data buffers. Can one be sure that all 16 channels are in one buffer. I.e., can one move this test outside of the element loop? (I recall trying to move it outside of the frequency loop and that seemed not OK, but not really sure any more.)